camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.builder;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.model.DataFormatDefinition;
+import org.apache.camel.model.transformer.CustomTransformerDefinition;
+import org.apache.camel.model.transformer.DataFormatTransformerDefinition;
+import org.apache.camel.model.transformer.EndpointTransformerDefinition;
+import org.apache.camel.model.transformer.TransformerDefinition;
+import org.apache.camel.spi.Transformer;
+
+/**
+ * A <a href="http://camel.apache.org/dsl.html">Java DSL</a> which is
+ * used to build a {@link org.apache.camel.spi.Transformer} and register into {@link org.apache.camel.CamelContext}.
+ */
+public class TransformerBuilder {
+
+    private String scheme;
+    private String from;
+    private String to;
+    private String uri;
+    private DataFormatDefinition dataFormat;
+    private Class<? extends Transformer> clazz;
+    private String beanRef;
+
+    /**
+     * Set a scheme name supported by the transformer.
+     *
+     * @param scheme scheme name
+     */
+    public TransformerBuilder scheme(String scheme) {
+        this.scheme = scheme;
+        return this;
+    }
+
+    /**
+     * Set the 'from' data type .
```

Comment:
cab you precise what is expected as "data type"? is it a classname? is a value taken from somewhere elese?